config/config.go

```diff
@@ -373,3 +385,27 @@ func (s *DestinationSubnet) GetRelayerAccountInfo() (*ecdsa.PrivateKey, common.A
 	pkBytes = append(pkBytes, pk.PublicKey.Y.Bytes()...)
 	return pk, common.BytesToAddress(crypto.Keccak256(pkBytes)), nil
 }
+
+//
+// Top-level config getters
+//
+
+// GetSourceIDs returns the Subnet and Chain IDs of all subnets configured as a source
+func (c *Config) GetSourceIDs() ([]ids.ID, []ids.ID, error) {
```

Comment:
It looks like this function is only called once from `main`, so it doesn't matter too much either way, but I would consider adding `sourceSubnetIDs` and `sourceChainIDs` to fields that are set once and then returned directly by `GetSourceIDs` rather than doing the iteration and construction each time it is called.

Reply:
Good call, added a caching step.